packages/Kokkos/src/DTK_LinearBVH.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef DTK_LINEAR_BVH_HPP
+#define DTK_LINEAR_BVH_HPP
+
+#include <Kokkos_ArithTraits.hpp>
+#include <Kokkos_Array.hpp>
+#include <Kokkos_Pair.hpp>
+#include <Kokkos_View.hpp>
+
+#include <Kokkos_OpenMP.hpp> // TMP
+
+namespace DataTransferKit
+{
+
+// Axis-Aligned Bounding Box
+// This is just a thin wrapper around an array of size 2x spatial dimension with
+// a default constructor to initialize properly an "empty" box.
+struct AABB
+{
+    using ArrayType = Kokkos::Array<double, 6>;
+    using SizeType = ArrayType::size_type;
+    AABB() = default;
+    AABB( ArrayType const &minmax )
+        : _minmax( minmax )
+    {
+    }
+    AABB &operator=( ArrayType const &minmax )
+    {
+        _minmax = minmax;
+        return *this;
+    }
+    double &operator[]( SizeType i ) { return _minmax[i]; }
+    double const &operator[]( SizeType i ) const { return _minmax[i]; }
+    ArrayType _minmax = {{
+        Kokkos::ArithTraits<double>::max(), -Kokkos::ArithTraits<double>::max(),
+        Kokkos::ArithTraits<double>::max(), -Kokkos::ArithTraits<double>::max(),
+        Kokkos::ArithTraits<double>::max(), -Kokkos::ArithTraits<double>::max(),
+    }};
+    friend std::ostream &operator<<( std::ostream &os, AABB const &aabb )
+    {
+        os << "{";
+        for ( int d = 0; d < 3; ++d )
+            os << " [" << aabb[2 * d + 0] << ", " << aabb[2 * d + 1] << "],";
+        os << "}";
+        return os;
+    }
+};
+struct Node
+{
+    virtual ~Node() = default;
+    Node *parent = nullptr;
+    Kokkos::pair<Node *, Node *> children = {nullptr, nullptr};
+    AABB bounding_box;
+};
+// Bounding Volume Hierarchy
+struct BVH
+{
+    using NodeType = Kokkos::OpenMP;
+    using DeviceType = NodeType::device_type;
+    BVH( AABB const *bounding_boxes, int n );
+    int size() const;
+    AABB bounds() const;
+    template <typename Predicate>
+    int query( Predicate const &predicates,
```

Comment:
The return value would be the number of collisions (extent/size of the `Kokkos::View`)